src/Type/Doctrine/QueryBuilder/OtherMethodQueryBuilderParser.php

```diff
@@ -51,18 +54,23 @@ public function findQueryBuilderTypesInCalledMethod(Scope $scope, MethodReflecti
 			return [];
 		}
 
+		$methodName = $methodReflection->getName();
 		$fileName = $methodReflection->getDeclaringClass()->getFileName();
 		if ($fileName === null) {
 			return [];
 		}
 
+		if (isset($this->cache[$fileName][$methodName])) {
+			return $this->cache[$fileName][$methodName];
```

Comment:
Note: if this gets proven to help, we should improve the cache key to support multiple classes with the same method name in the same file.